components/tools/OmeroJava/test/integration/gateway/TablesFacilityTest.java

```diff
@@ -150,6 +152,36 @@ public void testInvalidParams() throws Exception {
         } catch (Exception e) {
             // expected
         }
+        
+        try {
+            TableDataColumn[] columns = new TableDataColumn[1];
+            columns[0] = new TableDataColumn("column1", 0, ProjectData.class);
+            Object[][] data = new Object[1][1];
+            data[0][0] = new ProjectData();
+            TableData td = new TableData(columns, data);
+            tablesFacility.addTable(rootCtx, ds, "invalid", td);
+            Assert.fail("Invalid column type, an exception should have been thrown.");
+        } catch (Exception e) {
+            // expected
+        }
+    }
+
+    @Test(dependsOnMethods = { "testAddTable" })
+    public void testObjectColumnType() throws Exception {
```

Comment:
I probably should add a comment what this test is actually about. It basically checks that if there's no column type specified just general `Object.class`, a String column is created and the `Object.toString()` method is used (and a warning will be logged, although I can't really test that).